packages/template-chakra-storefront/src/components/with-translations/index.jsx

```diff
@@ -0,0 +1,122 @@
+ * - Eliminates repetitive useIntl() and formatMessage() calls in every component
+ * - Keeps JSX clean and readable without i18n clutter
+ * - Abstracts React Intl complexity away from component logic
+ * - Provides type-safe message references instead of magic strings
+ * - Enables static analysis for message extraction while maintaining clean syntax
+ * 
+ * @param {React.Component} Component - The React component to enhance with translations
+ * @param {Object} messageDefinitions - Message definitions object created with defineMessages()
+ * 
+ * @returns {React.Component} Enhanced component that receives a magical `messages` prop
+ * 
+ * @example
+ * // 1. Define messages using defineMessages
+ * const userMessages = defineMessages({
+ *   welcome: {
+ *     id: 'user.welcome',
+ *     defaultMessage: 'Welcome, {name}!'
+ *   },
+ *   title: {
+ *     id: 'user.title', 
+ *     defaultMessage: 'User Dashboard'
+ *   }
+ * });
+ * 
+ * // 2. Create pure component (no i18n imports needed)
+ * const UserDashboard = ({ messages, user }) => (
+ *   <div>
+ *     <h1>{messages.title}</h1>
+ *     <p>{messages.welcome({ name: user.name })}</p>
+ *   </div>
+ * );
+ * 
+ * // 3. Wrap component with HOC
+ * export default withTranslations(UserDashboard, userMessages);
+ * 
+ * MAGICAL MESSAGES API:
+ * The `messages` prop provides a magical object where each message key works in multiple ways:
+ * 
+ * Static usage (no parameters needed):
+ *   {messages.title} → "User Dashboard"
+ * 
+ * Dynamic usage (with interpolation):
+ *   {messages.welcome({ name: 'John' })} → "Welcome, John!"
+ *   {messages.welcome.format({ name: 'John' })} → "Welcome, John!" (alternative syntax)
+ * 
+ * JavaScript usage (for alerts, confirmations, etc.):
+ *   const msg = messages.welcome({ name: user.name });
+ *   alert(msg);
+ * 
+ * HOW THE MAGIC WORKS:
+ * - Uses JavaScript Proxy to intercept property access
+ * - Each message becomes a hybrid object that's both a string and a function
+ * - toString() and valueOf() methods provide automatic string conversion
+ * - Function interface allows dynamic parameter injection
+ * - Maintains all original message definition properties for compatibility
+ * 
+ * BENEFITS:
+ * ✅ Clean JSX: {messages.key} instead of {intl.formatMessage(...)}
+ * ✅ Single prop: Only need to pass `messages` to components
+ * ✅ Type-safe: Real message objects instead of string keys
+ * ✅ Component-controlled: Data for interpolation stays in the component
+ * ✅ Static analysis compatible: Works with React Intl extraction tools
+ * ✅ No imports: Components don't need to import React Intl
+ * ✅ Testable: Pure components are easy to unit test
+ * ✅ Reusable: Same HOC works with any message definitions
+ */
+export function withTranslations(Component, messageDefinitions) {
+  return function WrappedComponent(props) {
+    const intl = useIntl();
+    
+    // Create a magical messages object using Proxy
+    const messages = new Proxy({}, {
```

Comment:
Even though this is a small file, I don't think a new abstraction is really needed in this case. Using a Proxy I think adds complexity in terms of implementation. As for DX, it's a new interface that they need to learn, when they're already familiar with React Intl probably.